clang/lib/Sema/SemaBounds.cpp

```diff
@@ -705,6 +705,21 @@ namespace {
       // user when diagnosing unknown bounds errors.
       llvm::DenseMap<const VarDecl *, SmallVector<Expr *, 4>> UnknownSrcBounds;
 
+      // BlameAssignments maps a variable declaration V to an expression in a
+      // top-level CFG statement that last updates any variable used in the
+      // declared bounds of V.
+      //
+      // BlameAssignments is used to provide more context for two types of
+      // diagnostic messages:
+      //   1. The compiler cannot prove or can disprove the declared bounds for
```

Comment:
It is also possible that the compiler cannot prove or can disprove the declared bounds for V are valid after an assignment to a variable used in the declared bounds of V. For example:

```
unsigned int len;
array_ptr<int> p : count(len) = 0;
// Original value of len is len + 2
// Updated observed bounds of p are (p, p + len - 2)
// Cannot prove that bounds(p, p + len - 2) imply (p, p + len)
len = len + 2;
```

Reply:
I've changed the comment to this general case.